008-auth.md

```diff
@@ -0,0 +1,254 @@
+# Authentication and Authorization
+
+## Summary
+
+Right now, the generation of client authentication keys and RBACs thereof is a
+very manual process which is not easy to manipulate for the average user.
+Moreover, it does not encourage best practices such as having protection on
+private keys (via hardware tokens or passwords), shorter-duration certificates,
+user-oriented keys, or even the use of privilege-reduced operational certs.
+
+We should build some tooling to facilitate all of these needs and encourage best
+practices generally.
+
+Design goals:
+- simplify processes
+   - provide easy generation of client keypairs
+     - provide options for protecting client private keys:
+       - password-protection
+       - hardware keys
+   - provide easy generation of user certificates
+     - generate CSR from `talosctl`
+     - generate certificate from CSR using `talosctl`
+     - apply RBAC policies to certificates
+   - provide easy way to generate and update CRLs
+     - figure out how and where to house CRLs
+       - public service? (what if network is not available?)
+       - in etcd? (what if etcd is down?)
+       - on each node? (what if nodes are down during publish?)
+- greater authentication flexibility
+  - support external sources of authentication, such as OIDC servers (dex,
+    Google, etc)
+  - allow key pairs and certificates to be easily bound to managers instead of
+    roles
+- greater authorization flexibility
+  - allow decoupling roles from key sets
+  - allow dynamic and arbitrary mappings of authentication data to roles
+
+Related issues:
+ - https://github.com/talos-systems/talos/issues/3306
+
+## Case Studies
+
+### Any larger organisation
+
+In general, passing authentication credentials around on the network is an
+anti-pattern.
+Once the set of managers of a system grows beyond one, greater flexibility in
+the distribution and management of key and assignment material is necessary.
+
+### Organisations with an existing SSO system
+
+When an organisation already has in place a common authentication system,
+we should be able to externalise the sourcing of our authentication to their
+system.
+However, we still need to be able to control mappings of those identities to
```

Comment:
one way around that is to add additional accepted CAs to the Talos machine config and manage client certificate generation through that external system. so that Talos has break-the-glass CA which can always be used to access the system, but in general users do SSO via existing system to get (temporary) `talosconfig` to talk to the cluster